File: ch10/yojkim/ch10/franc.go
package ch10

type Franc struct {
	amount int
	currency string
}

func NewFranc(amount int) Franc {
	return Franc{
		amount: amount,
		currency: "CHF",
	}
}

func (f Franc) Times(multiplier int) Money {
	return NewFranc(f.amount * multiplier)
}

func (f Franc) Equal(input Money) bool {
	franc, ok := input.(Franc)
	if !ok {
		return false
	}
	return f.amount == franc.amount
}

func (f Franc) Currency() string {
	return f.currency
}

File: ch9/gunddol/src/ch9/Franc.java
package ch9;

public class Franc extends Money{

    protected String currency;

    public Franc(int amount, String currency){
        super(amount, currency);
    }

    Money times(int multiplier) {
        return Money.franc(amount * multiplier);
    }

//    String currency(){
//        return currency;
//    }
}
